src/frontend/js/components/SaleTunnelStepPayment/index.tsx

```diff
@@ -0,0 +1,275 @@
+    [creditCard, creditCards],
+  );
+
+  const selectedAddress = useMemo(
+    (): Maybe<Joanie.Address> =>
+      address ||
+      addresses.items.find((a) => a.is_main) ||
+      [...addresses.items].sort(sortAddressByTitleAsc)[0],
+    [address, addresses],
+  );
+
+  /**
+   * Memoized address items.
+   * If the selected address is a local one, we have to add it to the item list.
+   * Finally, we sort address ascending by title before return them
+   *
+   * @returns {Joanie.Address[]} Sorted addresses ascending by title
+   */
+  const addressesItems = useMemo(() => {
+    const items = [...addresses.items];
+
+    // Add local address to the address item list.
+    if (selectedAddress?.id === LOCAL_BILLING_ADDRESS_ID) {
+      items.push(selectedAddress);
+    }
+
+    return items.sort(sortAddressByTitleAsc);
+  }, [addresses.items, selectedAddress]);
+
+  /**
+   * Retrieve address through its `id` provided by the event target value
+   * then update `selectedAddress` state
+   *
+   * @param {React.ChangeEvent<HTMLSelectElement>} event
+   */
+  const handleSelectAddress = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const targetedAddress = addresses.items.find((a) => a.id === event.target.value);
+    if (targetedAddress) {
+      setAddress(targetedAddress);
+    }
+  };
+
+  const toggleCreditCard = (creditCardId: string) => {
+    if (selectedCreditCard === creditCardId) {
+      setCreditCard(null);
+    } else {
+      setCreditCard(creditCardId);
+    }
+  };
+
+  if (showAddressCreationForm) {
+    return (
+      <AddressesManagement
+        handleClose={() => setShowAddressCreationForm(false)}
+        selectAddress={setAddress}
+      />
+    );
+  }
+
+  if (!user) return null;
+
+  return (
+    <section className="SaleTunnelStepPayment">
+      <section className="SaleTunnelStepPayment__block">
+        <header className="SaleTunnelStepPayment__block__header">
+          <h5 className="SaleTunnelStepPayment__block__title">
+            <FormattedMessage {...messages.resumeTile} />
+          </h5>
+        </header>
+        <div className="SaleTunnelStepPayment__block--product">
+          <h6 className="SaleTunnelStepPayment__block--product__title">{product.title}</h6>
+          <p className="SaleTunnelStepPayment__block--product__price">
+            <FormattedNumber
+              value={product.price}
+              style="currency"
+              currency={product.price_currency}
+            />
+          </p>
+        </div>
+      </section>
+      <section className="SaleTunnelStepPayment__block">
+        <header className="SaleTunnelStepPayment__block__header">
+          <h5 className="SaleTunnelStepPayment__block__title">
+            <FormattedMessage {...messages.userTile} />
+          </h5>
+        </header>
+        <div className="SaleTunnelStepPayment__block--buyer">
+          <h6 className="SaleTunnelStepPayment__block--buyer__name">
```

Comment:
I'd also argue this is more content that something sectioning content and remove the heading tag here.